### ISSUE TITLE
Crashes on wrong frame size

### ISSUE THREAD

**Issue by uniqby** [chicio/ID3TagEditor#11]
**Describe the bug**
Crashes when trying to read some mp3 files

**To Reproduce**
1. Download http://cdndl.zaycev.net/807218/6982189/matrang_-_meduza_%28zaycev.net%29.mp3
2. Read tags from downloaded file

**Device (please complete the following information):**
 - Model: iPhone7 , iPhone X (emulator)
 - OS: iOs 12.1.1, iOs 11.0.1

**Additional context**
*** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[NSConcreteData subdataWithRange:]: range {113, 17764438} exceeds data length 8907127'
*** First throw call stack:
(
	0   CoreFoundation                      0x0000000104a331cb __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x0000000102eeff41 objc_exception_throw + 48
	2   CoreFoundation                      0x0000000104aa7b95 +[NSException raise:format:] + 197
	3   Foundation                          0x000000010255f0f6 -[NSData(NSData) subdataWithRange:] + 252
	4   ID3TagEditor                        0x0000000102ba0ee6 $S12ID3TagEditor0aB6ParserC12getFrameFrom33_1D3E9BDC72A711DCC87CD9BADE0624C3LL3mp38position7version10Foundation4DataVSo6NSDataC_SiAA0A7VersionOtF + 230
	5   ID3TagEditor                        0x0000000102ba0d05 $S12ID3TagEditor0aB6ParserC14parseFramesFor33_1D3E9BDC72A711DCC87CD9BADE0624C3LL3mp303id3B0ySo6NSDataC_AA0aB0CtF + 373
	6   ID3TagEditor                        0x0000000102ba07b7 $S12ID3TagEditor0aB6ParserC5parse3mp3AA0aB0CSg10Foundation4DataV_tKF + 551
	7   ID3TagEditor                        0x0000000102b9edd3 $S12ID3TagEditorAAC4read4fromAA0aB0CSgSS_tKF + 163
  ...
)
libc++abi.dylib: terminating with uncaught exception of type NSException

**Comment by chicio** [comment]
Hi @uniqby ,
thank you for your feedback. It seems I cannot open the file. Can you send me the mp3 file or a version of the mp3 file that contains only the id3 tag? In this way I can investigate the problem. 
Generally speaking if the frame size is wrong probably the entire id3 tag is corrupted. So the best approach could be to mark the id3 tag as corrupted and let ID3TagEditor return an exception.
Are you sure that the file passed to the ID3TagEditor was correct?

I will wait for your answer. In the meantime support my work by starring my github repositories 😃.
Thank you,
Fabrizio.

**Comment by uniqby** [comment]
Hi, @chicio 
thank you for reply!

The file is correct and i can open it in the https://picard.musicbrainz.org/ app.
<img width="857" alt="2018-12-12 0 28 13" src="https://user-images.githubusercontent.com/1488394/49831401-da13e180-fda4-11e8-8ab4-55aa41b50df1.png">

Link to the file: https://www.dropbox.com/s/etblwzo2xljv2ph/matrang_-_meduza_%28zaycev.net%29.mp3?dl=0

**Comment by chicio** [comment]
Hi @uniqby ,
I found the problem. I'm already working on the fix. You can start to see the changes and the description to have an idea of problem and the fix:

**WIP: Fix bug frame size for v2.4 #12**

I'm still working on the fix (now I fixed it in a brutal way only for the read part). I will let you know when everything is merged in master. This fix will be included in the release 2.2.0 (I'm waiting to publish it on cocoapods because there are some problem with the version 1.5.3 and Xcode 10, so I will update the release tag). 
Fabrizio.

**Comment by chicio** [comment]
Hi @uniqby ,
I merged the pull request I created with the fix. Inside the acceptance test I used your mp3 file to test the correctness of the implementation (anyway in the mp3 file I removed the music data just to avoid any legal problem 😉). So everything it's fixed and should work as expected. I'm closing the issue (let me know if you see any other problem).
Thank you and again continue to support my open source projects (I saw you starred everything, so thank you 😃 !!!!).
Fabrizio.
